knotx-core/knotx-api/src/main/java/com/cognifide/knotx/api/UriInfo.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Knot.x - Reactive microservice assembler
+ *
+ * Copyright (C) 2016 Cognifide Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.cognifide.knotx.api;
+
+public class UriInfo {
```

Comment:
All properties of `UriInfo` should be private and final as long only access to them is via getters and all are initialised in constructor.